### PR TITLE
qtstyleplugin-kvantum: use wrapQtAppsHook

### PR DESCRIPTION
**Pull request by Shados**
###### Motivation for this change
#65399

Shouldn't be needed for the style engine itself to work (I think), but it is needed for `kvantummanager` and `kvantumpreview`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @bugworm

### CODE REVIEW

**Comment by mmahut** [comment]
@Shados do you mind splitting that new package into a separate commit?

File: pkgs/development/libraries/qtstyleplugin-kvantum/default.nix.orig [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub, qmake, qtbase, qtsvg, qtx11extras, kwindowsystem, libX11, libXext, qttools }:
+
+stdenv.mkDerivation rec {
+  pname = "qtstyleplugin-kvantum";
+  version = "0.11.0";
```
Comment (mmahut) [inline pkgs/development/libraries/qtstyleplugin-kvantum/default.nix.orig]: Do you mind getting it to ``0.11.2``?

File: pkgs/development/libraries/qtstyleplugin-kvantum/default.nix.orig [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchFromGitHub {
+    owner = "tsujan";
+    repo = "Kvantum";
+    rev = "V${version}";
+    sha256 = "0aqlv7mj7r9vjm6pvb8pv3cyx35lgz8rjjas3k8wfdr3sqyyy25g";
+  };
+
+  nativeBuildInputs = [ qmake qttools ];
+  buildInputs = [ qtbase qtsvg qtx11extras kwindowsystem libX11 libXext  ];
```
Comment (mmahut) [inline pkgs/development/libraries/qtstyleplugin-kvantum/default.nix.orig]:
```suggestion
  buildInputs = [ qtbase qtsvg qtx11extras kwindowsystem libX11 libXext ];
```

**Comment by mmahut** [comment]
@GrahamcOfBorg build qtstyleplugin-kvantum

**Comment by Shados** [comment]
@mmahut there's no new package, just some weirdness on my end generated a spurious clone of the file o.0. I've fixed that and the noted style issue, and added a commit for the version bump to 0.11.2, which runs OK on my system.

**Comment by mmahut** [comment]
@GrahamcOfBorg build qtstyleplugin-kvantum

**Comment by mmahut** [comment]
@GrahamcOfBorg build qtstyleplugin-kvantum-qt4

**Comment by worldofpeace** [comment]
Backported in a465b1fa808d40d058f82261753d064678f14671 and c1090cc3e4461578d9190c22d4ebd25974d74f83. This was needed for the `wrapQtAppsHook` fixes.

Though a heads up: not using the custom deriver will break if https://github.com/NixOS/nixpkgs/pull/70691 is merged.

**Pull request merged**